tests/app/main/views/test_suppliers.py

```diff
@@ -1102,35 +1139,45 @@ def test_should_404_if_agreement_not_returned(self, s3, data_api_client):
         data_api_client.get_framework.assert_called_with('g-cloud-8')
         data_api_client.get_supplier_framework_info.assert_called_with('1234', 'g-cloud-8')
 
+    def test_should_404_if_agreement_has_no_version(self, s3, data_api_client):
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.get_framework.return_value = {'frameworks': {}}
+        response = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
+
+        assert response.status_code == 404
+        data_api_client.get_supplier.assert_called_with('1234')
+        data_api_client.get_framework.assert_called_with('g-cloud-8')
+
+    def _find_services_iter_side_effect(self, *args, **kwargs):
```

Comment:
I'd be tempted to reqrite this as
```
data_api_client.find_services_iter.return_value = iter((
    {"id": 1111, "lotSlug": "dried-fruit", "lotName": "Raisins & dates"},
    {"id": 2222, "lotSlug": "salad", "lotName": "Lettuce & cucumber"},
    {"id": 3333, "lotSlug": "dried-fruit", "lotName": "Raisins & dates"}
))
```
In the actual test method. Those assertions don't really belong there or add any value and if we're not going to have them then we can avoid having a function or method all together.

Reply:
Good call. I admit I didn't look at what this test was actually doing too closely, just moved it around. Will fix.